src/particles/profiles/GetInitialDensity.H

```diff
@@ -42,13 +43,17 @@ struct GetInitialDensity
                                 * exp( -0.5_rt*delta_z*delta_z );
         } else if (m_profile == BeamProfileType::Flattop) {
             density = m_density;
+        } else if (m_profile == BeamProfileType::Parsed) {
+            density = m_density_func(x, y, z);
         }
         return density;
     }
     amrex::RealVect  m_position_mean {0.,0.,0.}; /* mean in case of a Gaussian density distribution. */
     amrex::RealVect  m_position_std {0.,0.,0.};  /* rms standard deviation in case of a Gaussian density distribution */
     BeamProfileType m_profile; /* beam profile type, e.g. BeamProfileType::Flattop or BeamProfileType::Gaussian*/
     amrex::Real m_density; /**< Peak density of the beam */
+    amrex::Parser m_parser; /**< owns data for m_density_func */
```

Comment:
m_parser can not be inside this struct because it is transferred to the GPU and m_parser owns the data.